packages/node/src/handlers.ts

```diff
@@ -260,11 +254,28 @@ export function parseRequest(
     event.server_name = global.process.env.SENTRY_NAME || os.hostname();
   }
 
-  if (options.user && req.user) {
-    event.user = {
-      ...event.user,
-      ...extractUserData(req, options.user),
-    };
+  if (options.user) {
+    const extractedUser = req.user ? extractUserData(req.user, options.user) : {};
+
+    if (Object.keys(extractedUser)) {
+      event.user = {
+        ...event.user,
+        ...extractedUser,
+      };
```

Comment:
By doing them in this order, is it possible we'd be overwriting user data added using `withScope` with the extracted data?

Reply:
Yes, but this was the provious behavior and I didn't want to change it here.

Reply:
I ask because it just came up as a bug on the Python side (see https://github.com/getsentry/sentry-python/pull/637). Even if we don't do it here (if we want to consider it a breaking change), might be worth thinking about as a way to prevent future user confusion/frustration.

Reply:
Agree. Let's get this in first, so it allows for extraction at all. It won't do any more harm in this form anyway.

Reply:
Sounds good.

Reply:
Created an issue so we don't forget :-)

https://github.com/getsentry/sentry-javascript/issues/2472